designing-reports/report-designer-tools/web-report-designer/tools/jsondatasource-wizard.md

```diff
@@ -0,0 +1,27 @@
+---
+title: JsonDataSource Wizard
+page_title: JsonDataSource Wizard | for Telerik Reporting Documentation
+description: jsonDataSource Wizard
+slug: telerikreporting/designing-reports/report-designer-tools/web-report-designer/tools/jsondatasource-wizard
+tags: jsondatasource,wizard
+published: True
+position: 2
+---
+
+# JsonDataSource Wizard
+
+The Json Data Source Wizard allows you to create a new or edit an existing JsonDataSource component in the Telerik Web Report Designer. This article describes the steps you can follow after the wizard appears, after being invoked from the Report Designer toolbox. 
+
+1. __Choose a JSON source__ Select how the json source should be retrieved for the component. 
+
+   + __Upload JSON file__ from the local computer                 
+
+   + __Use existing JSON file__ that previously has been uploaded and/or exists in the [IResourceStorage](/reporting/api/Telerik.WebReportDesigner.Services.IResourceStorage) The default implementation of the [IResourceStorage](/reporting/api/Telerik.WebReportDesigner.Services.IResourceStorage) interface - [ResourceStorage](/reporting/api/Telerik.WebReportDesigner.Services.ResourceStorage) stores resources like json files to the location provided as a string parameter in the constructor [#ctor](/reporting/api/Telerik.WebReportDesigner.Services.ResourceStorage#Telerik_WebReportDesigner_Services_ResourceStorage_#ctor_System_String_). When choosing the option __"Use existing json file"__, any existing json files found in this directory will populate the dropdown list. 
+
+   + __Use JSON inline__ to paste in a string in the text area 
+
+Optionally, you can use the __Data selector__ to query and filter the JSON data.The data selector is a JSONPath string which will be used to query the data. 
```

Comment:
'JSON data.The data selector is a JSONPath string': Insert a space after the full stop. I would also make JSONPath a link to "https://goessner.net/articles/JsonPath/"